cmd/translate/main.go

```diff
@@ -122,27 +124,33 @@ func main() {
 func init() {
 	cobra.OnInitialize(initConfig)
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is ./translate.yaml)")
+	rootCmd.PersistentFlags().Uint("port", 8080, "port to run on") //nolint:gomnd
 }
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if cfgFile != "" {
-		// Use config file from the flag.
-		viper.SetConfigFile(cfgFile)
-	} else {
-		// Find current dir.
-		dir, err := os.Getwd()
-		cobra.CheckErr(err)
-
-		// Search config in current directory with name "translate.yaml".
-		viper.AddConfigPath(dir)
-		viper.SetConfigFile("translate.yaml")
+	// Use default config file if not set.
+	if cfgFile == "" {
+		cfgFile = "translate.yaml"
 	}
 
-	viper.AutomaticEnv() // read in environment variables that match
+	viper.SetConfigFile(cfgFile)
+
+	viper.SetEnvPrefix("translate")
+	// Replace underscores with dots in environment variable names.
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+
+	// Try to read config.
+	if err := viper.ReadInConfig(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	} else {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
 
-	// If a config file is found, read it in.
-	if err := viper.ReadInConfig(); err == nil {
-		fmt.Fprintln(os.Stderr, "Using config file:", viper.ConfigFileUsed())
+	// For now manually bind CLI argument to viper.
+	err := viper.BindPFlag("service.port", rootCmd.Flags().Lookup("port"))
+	if err != nil {
```

Comment:
I am open to suggestions, what would be the best way to bind arguments, from options I found